File: src/data/courses/dsa/two-pointers-sliding-window/sliding-window-variable-shortest/minimum-size-subarray-sum/code/python/minimum_size_subarray_sum_sol1.py
class Solution:
    def minSubArrayLen(self, target, nums):
        min_len = float('inf')
        for i in range(len(nums)):
            s = 0
            for j in range(i, len(nums)):
                s += nums[j]
                if s >= target: min_len = min(min_len, j-i+1); break
        return 0 if min_len == float('inf') else min_len

File: src/data/courses/dsa/two-pointers-sliding-window/sliding-window-variable-shortest/minimum-size-subarray-sum/code/java/MinimumSizeSubarraySumSol1.java
class Solution {
    public int minSubArrayLen(int target, int[] nums) {
        int min = Integer.MAX_VALUE;
        for (int i=0;i<nums.length;i++) {
            int sum = 0;
            for (int j=i;j<nums.length;j++) { sum+=nums[j]; if(sum>=target){min=Math.min(min,j-i+1);break;} }
        } return min==Integer.MAX_VALUE?0:min;
    }
}